#! /bin/sh
#  ## Software required for running the various bash script modules
#  
#  The following software is necessary for running all scripts along with R packages found in the scripts in th 5_statistical_analysis sub-folders
#  
#  [perl v5.32](https://metacpan.org/release/perl)
#  
#  [Edirect tools v14.2](https://dataguide.nlm.nih.gov/edirect/install.html)
#  
#  [SRA toolkit v2.10.9](https://github.com/ncbi/sra-tools/wiki/02.-Installing-SRA-Toolkit)
#  
#  [Flexbar v3.5](https://github.com/seqan/flexbar)
#  
#  ### 1) Download and merge data in order to mimic the analysis performed in the mansucript
#  
#  
#  Install the Edirect tools v14.2 according to the instructions of https://dataguide.nlm.nih.gov/edirect/install.html
#  
for MICROBIOME in 1_tbz_vs_suc 2_tbz13C_vs_tbz12C
do
	cd 1_fetch_and_demultiplex/${MICROBIOME}/1_raw_sequences
	#  ```
	#  Install the SRA toolkit v2.10.9 according to the instructions of https://github.com/ncbi/sra-tools/wiki/02.-Installing-SRA-Toolkit
	#  
	#  Then, download the fastq in split forward/reverse format
	#  ```
	for i in `cat SRR_accessions.txt`
	do
		fastq-dump -I --split-files ${i}
	done
	#  ```
	#  Concatenate the files to a single library parted by a generic forward reads file and a generic reverse reads file to mimic the files generated by the sequencing center and store them in the 2_demultiplex_data folder
	#  ```
	cat *_1.fastq > ../2_demultiplexed_sequences/R1.fastq
	cat *_2.fastq > ../2_demultiplexed_sequences/R2.fastq

	#  ```
	#  ### 2) Demultiplex data using the script prepared by Sotirios Vasileiadis which is based on the [Flexbar v3.5 software](https://github.com/seqan/flexbar)
	#  Start with the prokaryotes
	#  ```
	cd ../2_demultiplexed_sequences
	#  ```
	#  You can test the "DemuxOwnBCsys_absPATH.sh" for necessary arguments with "sh DemuxOwnBCsys_absPATH.sh"
	#  Run the demultiplexing process by calling the "run_demux.sh" bash script which is created by the run_demux_pre.sh script after replacing the absolute path value with the current directory path 
	#  ```
	cp ../../scripts/*.sh ./
	perl -pe "s|absolutepath|\\Q${PWD}|g" run_demux_pre.sh | perl -pe "s/\\\//g" > run_demux.sh
	#  ```
	#  Attention!!! the generated last two files (last index read pairs) will remain gnu-zipped. This is a minor pipeline glitch and you need to unzip them manually or add a script.
	#  ```
	sh run_demux.sh

	cd ../../..
done
